# Report for ForestQuery into Global Deforestation, 1990 to 2016

ForestQuery is on a mission to combat deforestation around the world and to raise awareness about this topic and its impact on the environment. The data analysis team at ForestQuery has obtained data from the World Bank that includes forest area and total land area by country and year from 1990 to 2016, as well as a table of countries and the regions to which they belong.

The data analysis team has used SQL to bring these tables together and to query them in an effort to find areas of concern as well as areas that present an opportunity to learn from successes.


In [1]:
print("Executing postgresql setup...")
%reload_ext sql
%sql postgresql://jadug.parusa:@localhost/deforestation
%sql SELECT * FROM forest_area WHERE forest_area_sqkm>0 ORDER BY 3 DESC, 4 DESC LIMIT 3

Executing postgresql setup...
 * postgresql://jadug.parusa:***@localhost/deforestation
3 rows affected.


[('WLD', 'World', 2016, Decimal('39958245.9')),
 ('RUS', 'Russian Federation', 2016, Decimal('8148895')),
 ('BRA', 'Brazil', 2016, Decimal('4925540'))]

In [2]:
%%sql

-- #DROP VIEW IF EXISTS deforestation_tab;
CREATE OR REPLACE VIEW deforestation_tab AS (
    SELECT
        f.country_code,
        f.country_name,
        f.year,
        f.forest_area_sqkm,
        (2.59 * l.total_area_sqmi) AS total_area_sqkm,
        r.region,
        r.income_group
    FROM forest_area AS f
    JOIN
        land_area AS l
        ON
            f.country_code = l.country_code
        AND
            f.year = l.year
    JOIN
        regions AS r
        ON
            f.country_code = r.country_code
);
SELECT *, COUNT(*) over()
FROM deforestation_tab
WHERE forest_area_sqkm>0
GROUP BY 1,2,3,4,5,6,7
ORDER BY 3 DESC, 4 DESC
LIMIT 3;

 * postgresql://jadug.parusa:***@localhost/deforestation
Done.
3 rows affected.


[('WLD', 'World', 2016, Decimal('39958245.9'), Decimal('127354641.4357'), 'World', 'NULL', 5570),
 ('RUS', 'Russian Federation', 2016, Decimal('8148895'), Decimal('16376869.9997'), 'Europe & Central Asia', 'Upper middle income', 5570),
 ('BRA', 'Brazil', 2016, Decimal('4925540'), Decimal('8358139.9972'), 'Latin America & Caribbean', 'Upper middle income', 5570)]

## 1. GLOBAL SITUATION

#### Question 1.1: How was the World's 1990 total forest area compared to 2016?

In [3]:
%%sql
-- # filter country (world), year (1990 or 2016), calculate (area, difference, and percentage)
DROP VIEW IF EXISTS world_forest_loss;
CREATE VIEW world_forest_loss AS (
    WITH
    world_1990_2016 AS (
        SELECT
            country_name,
            year,
            forest_area_sqkm
        FROM
            deforestation_tab
        WHERE
            country_name = 'World'
            AND (
                year = 1990
                OR year = 2016
            )
    ),
    forest_1990 AS (
        SELECT
            forest_area_sqkm AS base_1990_sqkm
        FROM
            world_1990_2016
        WHERE
            year = 1990
    )

    SELECT
        t.country_name,
        t.year,
        t.forest_area_sqkm,
        base_1990_sqkm,
        (t.forest_area_sqkm - base_1990_sqkm) AS loss_sqkm,
        100*ROUND((t.forest_area_sqkm - base_1990_sqkm) / base_1990_sqkm, 3) AS loss_percent
    FROM
        world_1990_2016 AS t,
        forest_1990
    ORDER BY 2
);

SELECT * FROM world_forest_loss

 * postgresql://jadug.parusa:***@localhost/deforestation
Done.
Done.
2 rows affected.


[('World', 1990, Decimal('41282694.9'), Decimal('41282694.9'), Decimal('0.0'), Decimal('0.000')),
 ('World', 2016, Decimal('39958245.9'), Decimal('41282694.9'), Decimal('-1324449.0'), Decimal('-3.200'))]

#### Question 1.2: How comparable is this forest area loss against a single country size?

In [4]:
%%sql
-- # filter year (2016), filter and order (by total_area_sqkm)
WITH base_2016_loss AS (
    SELECT
        loss_sqkm AS world_loss_sqkm
    FROM
        world_forest_loss
    WHERE
        year = 2016
)

SELECT
    *
FROM
    deforestation_tab AS d,
    base_2016_loss
WHERE
    year = 2016
    AND d.total_area_sqkm < -(world_loss_sqkm)
ORDER BY 5 DESC
LIMIT 3

 * postgresql://jadug.parusa:***@localhost/deforestation
3 rows affected.


[('PER', 'Peru', 2016, Decimal('738053.9844'), Decimal('1279999.9891'), 'Latin America & Caribbean', 'Upper middle income', Decimal('-1324449.0')),
 ('NER', 'Niger', 2016, Decimal('11295.99976'), Decimal('1266700.0024'), 'Sub-Saharan Africa', 'Low income', Decimal('-1324449.0')),
 ('TCD', 'Chad', 2016, Decimal('47483.99902'), Decimal('1259200.0099'), 'Sub-Saharan Africa', 'Low income', Decimal('-1324449.0'))]

According to the World Bank, the total forest area of the world was `41,282,694.9 sqkm` in 1990. As of 2016, the most recent year for which data was available, that number had fallen to `39,958,245.9 sqkm`, a loss of `-1,324,449.0 sqkm`, or `-3.200%`.

The forest area lost over this time period is slightly more than the entire land area of `Peru` listed for the year 2016 (which is `1,279,999.9891 sqkm`).


## 2. REGIONAL OUTLOOK

#### Question 2.1:
How many percent of World's land area was designated as forest, in 2016 and 1990?

Which region had the highest & the lowest forestation percentage respectively in 2016 and 1990?

In [5]:
%%sql
-- # filter year (1990 or 2016), calculate (forest area, land area, and percentage),
-- # select world's, highest region, lowest region
-- # apply union
DROP VIEW IF EXISTS region_forestation;
CREATE VIEW region_forestation AS (
    SELECT
        region,
        year,
        ROUND(100 * SUM(forest_area_sqkm) / SUM(total_area_sqkm), 3) AS forestation_percentage,
        ROUND(SUM(forest_area_sqkm),3) AS region_forest_sqkm,
        ROUND(SUM(total_area_sqkm),3) AS region_land_sqkm
    FROM
        deforestation_tab
    WHERE
        year = 2016
        OR year = 1990
        AND forest_area_sqkm > 0 -- excluding countries having empty forest_area_sqkm
        AND total_area_sqkm > 0 -- excluding countries having empty total_area_sqkm
    GROUP BY 1,2
    ORDER BY 2, 3 DESC
);
WITH
world_forestation AS (
    SELECT * FROM region_forestation WHERE region = 'World' LIMIT 2
),
highest_2016 AS (
    SELECT * FROM region_forestation WHERE year = 2016 ORDER BY 3 DESC LIMIT 1
),
lowest_2016 AS (
    SELECT * FROM region_forestation WHERE year = 2016 ORDER BY 3 LIMIT 1
),
highest_1990 AS (
    SELECT * FROM region_forestation WHERE year = 1990 ORDER BY 3 DESC LIMIT 1
),
lowest_1990 AS (
    SELECT * FROM region_forestation WHERE year = 1990 ORDER BY 3 LIMIT 1
)

SELECT * FROM world_forestation
UNION
SELECT * FROM highest_2016
UNION
SELECT * FROM lowest_2016
UNION
SELECT * FROM highest_1990
UNION
SELECT * FROM lowest_1990

ORDER BY 1 DESC,2

 * postgresql://jadug.parusa:***@localhost/deforestation
Done.
Done.
6 rows affected.


[('World', 1990, Decimal('32.422'), Decimal('41282694.900'), Decimal('127328467.440')),
 ('World', 2016, Decimal('31.376'), Decimal('39958245.900'), Decimal('127354641.436')),
 ('Middle East & North Africa', 1990, Decimal('1.777'), Decimal('199292.596'), Decimal('11214619.995')),
 ('Middle East & North Africa', 2016, Decimal('2.068'), Decimal('232131.004'), Decimal('11223465.985')),
 ('Latin America & Caribbean', 1990, Decimal('51.031'), Decimal('10242341.796'), Decimal('20070746.441')),
 ('Latin America & Caribbean', 2016, Decimal('46.162'), Decimal('9250585.884'), Decimal('20039364.447'))]

In 2016, the percent of the total land area of the world designated as forest was `31.376%`. The region with the highest relative forestation was `Latin America & Caribbean`, with `46.162%`, and the region with the lowest relative forestation was `Middle East & North Africa`, with `2.068%` forestation.

In 1990, the percent of the total land area of the world designated as forest was `32.422%`. The region with the highest relative forestation was `Latin America & Caribbean`, with `51.031%`, and the region with the lowest relative forestation was `Middle East & North Africa`, with `1.777%` forestation.


#### Question 2.2:
From 1990 to 2016, which regions had increasing forestation & which regions had decreasing forestation?

How did those regional increase & decrease impact to the overall World' forestation?

In [6]:
%%sql
-- # self join to combine percentage (1990 or 2016)
WITH
rf_selfjoin AS (
    SELECT
        rf_1990.region,
        rf_1990.forestation_percentage AS forestation_1990,
        rf_2016.forestation_percentage AS forestation_2016
    FROM region_forestation AS rf_1990
    JOIN region_forestation AS rf_2016
    ON
        rf_1990.region = rf_2016.region
        AND rf_1990.year = 1990
        AND rf_2016.year = 2016
)

SELECT
    region,
    forestation_1990,
    forestation_2016,
    (forestation_2016 - forestation_1990) AS net_percentage_increase
FROM
    rf_selfjoin
ORDER BY 4

 * postgresql://jadug.parusa:***@localhost/deforestation
8 rows affected.


[('Latin America & Caribbean', Decimal('51.031'), Decimal('46.162'), Decimal('-4.869')),
 ('Sub-Saharan Africa', Decimal('32.190'), Decimal('28.788'), Decimal('-3.402')),
 ('World', Decimal('32.422'), Decimal('31.376'), Decimal('-1.046')),
 ('Middle East & North Africa', Decimal('1.777'), Decimal('2.068'), Decimal('0.291')),
 ('North America', Decimal('35.651'), Decimal('36.039'), Decimal('0.388')),
 ('East Asia & Pacific', Decimal('25.771'), Decimal('26.359'), Decimal('0.588')),
 ('Europe & Central Asia', Decimal('37.271'), Decimal('38.041'), Decimal('0.770')),
 ('South Asia', Decimal('16.511'), Decimal('17.506'), Decimal('0.995'))]

The only regions of the world that decreased in percent forest area from 1990 to 2016 were `Latin America & Caribbean` (dropped from `51.031%` to `46.162%`) and `Sub-Saharan Africa` (`32.190%` to `28.788%`).

All other regions actually increased in forest area over this time period. However, the drop in forest area in the two aforementioned regions was so large, the percent forest area of the world decreased over this time period from `32.422%` to `31.376%`. 


## 3. COUNTRY-LEVEL DETAIL

### A. SUCCESS STORIES

There is one particularly bright spot in the data at the country level, __________________. This country actually increased in forest area from 1990 to 2016 by __________________. It would be interesting to study what has changed in this country over this time to drive this figure in the data higher. The country with the next largest increase in forest area from 1990 to 2016 was the__________________, but it only saw an increase of __________________, much lower than the figure for __________________.

__________________ and __________________ are of course very large countries in total land area, so when we look at the largest percent change in forest area from 1990 to 2016, we aren’t surprised to find a much smaller country listed at the top. __________________ increased in forest area by __________________% from 1990 to 2016. 


### B. LARGEST CONCERNS

Which countries are seeing deforestation to the largest degree? We can answer this question in two ways. First, we can look at the absolute square kilometer decrease in forest area from 1990 to 2016. The following 3 countries had the largest decrease in forest area over the time period under consideration:

The second way to consider which countries are of concern is to analyze the data by percent decrease.

When we consider countries that decreased in forest area percentage the most between 1990 and 2016, we find that four of the top 5 countries on the list are in the region of __________________. The countries are __________________, __________________, __________________, and __________________. The 5th country on the list is __________________, which is in the __________________ region. 

From the above analysis, we see that __________________ is the only country that ranks in the top 5 both in terms of absolute square kilometer decrease in forest as well as percent decrease in forest area from 1990 to 2016. Therefore, this country has a significant opportunity ahead to stop the decline and hopefully spearhead remedial efforts.


### C. QUARTILES

The largest number of countries in 2016 were found in the __________________ quartile.

There were __________________ countries in the top quartile in 2016. These are countries with a very high percentage of their land area designated as forest. The following is a list of countries and their respective forest land, denoted as a percentage.


## 4. RECOMMENDATIONS